rdbms/src/main/java/org/spine3/server/storage/jdbc/util/DataSourceWrapper.java

```diff
@@ -61,8 +67,25 @@ public ConnectionWrapper getConnection(boolean autoCommit) throws DatabaseExcept
         }
     }
 
+    /**
+     * Closes wrapped {@link DataSource} implementation if it implements {@link AutoCloseable}.
+     * Otherwise warning will be thrown.
+     *
+     * @throws DatabaseException
+    */
     @Override
-    public abstract void close() throws DatabaseException;
+    public void close() throws DatabaseException{
+        if (dataSource instanceof AutoCloseable){
```

Comment:
And don't forget to add spaces while you are typing, this is just a simple habit)